dffml/service/dev.py

```diff
@@ -1056,6 +1060,96 @@ class Bump(CMD):
     packages = BumpPackages
 
 
+class SphinxBuildError(Exception):
+    """
+    Raised when sphinx-build command exits with a non-zero error code.
+    """
+
+    def __str__(self):
+        return "[ERROR] Failed run sphinx, is it installed (pip install -U .[dev])?"
+
+
+class MakeDocs(CMD):
+    """
+    Generate docs of the complete package 
+    """
+
+    async def run(self):
+
+        root = Path(__file__).parents[2]
+
+        pages_path = root / "pages"
+
+        shutil.rmtree(pages_path, ignore_errors=True)
+
+        docs_path = root / "docs"
+
+        files_to_check = [
+            ("changelog.md", "CHANGELOG.md"),
+            ("shouldi.md", "examples/shouldi/README.md"),
+            ("swportal.rst", "examples/swportal/README.rst"),
+            (
+                "contributing/consoletest.md",
+                "dffml/util/testing/consoletest/README.md",
+            ),
+        ]
+
+        for symlink, source in files_to_check:
+            file_path = docs_path / symlink
+            if not file_path.exists():
+                file_path.symlink_to(root / source)
+
+        # HTTP Service Docs
+        service_path = docs_path / "plugins/service"
+        service_path.mkdir(parents=True, exist_ok=True)
+        http_docs_pth = service_path / "http"
+        os.unlink(http_docs_pth)
+        http_docs_pth.symlink_to(root / "service/http/docs")
+
+        # Main Docs
+        runpy.run_path(root / "scripts/docs.py")
+        runpy.run_path(root / "scripts/docs_api.py")
+
+        cmd = ["sphinx-build", "-W", "-b", "html", "docs", "pages"]
+        print(f"$ {' '.join(cmd)}")
+        proc = await asyncio.create_subprocess_exec(*cmd)
+        await proc.wait()
+        if proc.returncode != 0:
+            raise SphinxBuildError
+
+        images_path = docs_path / "images"
+        images_set = set(images_path.iterdir())
+        target_set = set(pages_path.iterdir())
+
+        files_to_copy = list(images_set - target_set)
+        for file in files_to_copy:
+            shutil.copy(file, pages_path)
+
+        copybutton_path = pages_path / "_static/copybutton.js"
+
+        with cached_download(
+            "https://raw.githubusercontent.com/python/python-docs-theme/master/python_docs_theme/static/copybutton.js",
+            copybutton_path,
+            "061b550f64fb65ccb73fbe61ce15f49c17bc5f30737f42bf3c9481c89f7996d0004a11bf283d6bd26cf0b65130fc1d4b",
+        ) as _:
+            pass
+
+        nojekyll_path = pages_path / ".nojekyll"
+        nojekyll_path.touch(exist_ok=True)
+
+        if os.environ.get("HTTP") is not None:
+            PORT = 8080
+
+            class Handler(http.server.SimpleHTTPRequestHandler):
+                def __init__(self, *args, **kwargs):
+                    super().__init__(
+                        *args, directory=str(pages_path), **kwargs
+                    )
```

Comment:
> https://github.com/intel/dffml/commit/04401c28878c61f941e92daf89d7c3056b6eb77c#diff-5f4a6a6b533ec96b5895bc79efc24910ae3e9c109d653717ec77474194faf1f7R617

Copying the way the stdlib does it.

```suggestion
            Handler = functools.partial(http.server.SimpleHTTPRequestHandler, directory=str(pages_path))
```

Reply:
Done